### PR TITLE
blog: Deploying FlowFuse with Docker on the Ubuntu server

### PR DESCRIPTION
**Pull request by sumitshinde-84**
## Checklist

 - [ ] I have read the [contribution guidelines](https://github.com/FlowFuse/flowfuse/blob/main/CONTRIBUTING.md)
 - [ ] I have considered the performance impact of these changes
 - [ ] Suitable unit/system level tests have been added and they pass
 - [ ] Documentation has been updated

### CODE REVIEW

File: src/blog/2024/07/deploying-flowfuse-with-docker.md [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+
+```bash
+ sudo apt-get install docker-ce docker-ce-cli containerd.io docker-buildx-plugin docker-compose-plugin
+```
+
+```bash
+sudo docker run hello-world
+```
+
+Now, if you see the result similar to the below image it means the docker is installed successfully.
```
Comment (gdziuba) [inline src/blog/2024/07/deploying-flowfuse-with-docker.md:70]:
```suggestion
If installed correctly, you should see results similar to the image below.
```

File: src/blog/2024/07/deploying-flowfuse-with-docker.md [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+
+!["Screenshot the terminal showing the updates made in 'docker-compose.yml' file"](./images/docker-compose-update-2.png "Screenshot the terminal showing the updates made in 'docker-compose.yml' file")
+
+For your understanding, We updated the `flowforge.yml` file to include our domain in the following fields: `domain`, `base_url`, and `broker.public_url`. These changes ensure that instance names on Docker platforms prepend your domain, provide the correct URL to access the platform, and specify the URL for devices to connect to the broker if different from `broker.url`.
+
+Additionally, in the docker-compose.yml file, we adjusted the `VIRTUAL_HOST` to point to our domain.
+
+### Securing Communication with SSL
+
+Securing communication with [SSL (Secure Sockets Layer)](https://www.youtube.com/watch?v=SJJmoDZ3il8) is crucial for protecting data transmitted between your users and the server. Adding SSL requires obtaining a certificate from a trusted certificate authority (CA) and configuring your server to use this certificate. Configuring SSL manually can be a headache, so we have provided a setup that you just need to enable.
```
Comment (gdziuba) [inline src/blog/2024/07/deploying-flowfuse-with-docker.md:136]:
```suggestion
Securing communication with [SSL (Secure Sockets Layer)](https://www.youtube.com/watch?v=SJJmoDZ3il8) is crucial for protecting data transmitted between your users and the server. Adding SSL requires obtaining a certificate from a trusted certificate authority (CA) and configuring your server to use this certificate. Configuring SSL manually can be a headache, so we have provided a setup you need to enable.
```

File: src/blog/2024/07/deploying-flowfuse-with-docker.md [hunk 3]
```diff
@@ -0,0 +1,300 @@
+---
+title: Deploying FlowFuse with Docker on the Ubuntu server
```
Comment (gdziuba) [inline src/blog/2024/07/deploying-flowfuse-with-docker.md:2]:
```suggestion
title: Deploying FlowFuse with Docker on an Ubuntu server
```

File: src/blog/2024/07/deploying-flowfuse-with-docker.md [hunk 4]
```diff
@@ -0,0 +1,300 @@
...
+subtitle: Step-by-step guide on how to Deploy FlowFuse with FlowFuse with Docker on Ubuntu server
```
Comment (gdziuba) [inline src/blog/2024/07/deploying-flowfuse-with-docker.md:3]:
```suggestion
subtitle: Step-by-step guide on how to deploy FlowFuse with Docker on Ubuntu server
```

**Review by gdziuba — CHANGES_REQUESTED** [review]

**Review by Hasmin-AC — APPROVED** [review]

File: src/blog/2024/07/deploying-flowfuse-with-docker.md [hunk 5]
```diff
@@ -0,0 +1,300 @@
+---
+title: Deploying FlowFuse with Docker on an Ubuntu server
+subtitle: Step-by-step guide on how to deploy FlowFuse with Docker on Ubuntu server
+description: Learn to deploy the FlowFuse on your Ubuntu server with Docker, from domain setup to running your app with email, SSL configurations.
+date: 2024-07-08
+authors: ["sumit-shinde"]
+image:
```
Comment (gdziuba) [inline src/blog/2024/07/deploying-flowfuse-with-docker.md:7]: image?

File: src/blog/2024/07/deploying-flowfuse-with-docker.md [hunk 6]
```diff
@@ -0,0 +1,300 @@
...
+
+## What is Docker?
+
+[Docker](https://docs.docker.com/guides/docker-concepts/the-basics/what-is-an-image/) is an open-source platform that simplifies how applications are deployed, scaled, and managed through containerization. It enables you to package all components required by your project such as code, libraries, and dependencies into a single, portable unit known as a [Docker container](https://docs.docker.com/guides/docker-concepts/the-basics/what-is-a-container/). These containers ensure consistency in application environments and ease deployment by ensuring that applications run predictably across different computing environments, whether on a developer's laptop, a server, or a cloud platform.
+
+## Deploying FlowFuse on Ubuntu server with Docker
+
+Before proceeding, ensure you have your domain and a server with Ubuntu installed. If your server has Ubuntu installed without a GUI, you can connect to it from your computer using [SSH](https://itsfoss.com/set-up-ssh-ubuntu/), so that you can run commands from your local computer on the server.
+
+### Installing Docker on the Ubuntu server.
```
Comment (gdziuba) [inline src/blog/2024/07/deploying-flowfuse-with-docker.md:27]: Should we have documentation on our website on how to install other company's products?  Shouldn't this be a reference?

Comment (gdziuba) [reply src/blog/2024/07/deploying-flowfuse-with-docker.md:27]: This is in reference to Docker.

Comment (sumitshinde-84) [reply src/blog/2024/07/deploying-flowfuse-with-docker.md]: Removed that section and added a link to their docs.

**Comment by sumitshinde-84** [comment]
Tile has been added now !

**Review by gdziuba — APPROVED** [review]

File: src/blog/2024/07/deploying-flowfuse-with-docker.md [hunk 7]
```diff
@@ -0,0 +1,253 @@
+---
+title: Deploying FlowFuse with Docker on an Ubuntu server
+subtitle: Step-by-step guide on how to deploy FlowFuse with Docker on Ubuntu server
+description: Learn to deploy the FlowFuse on your Ubuntu server with Docker, from domain setup to running your app with email, SSL configurations.
+date: 2024-07-15
```
Comment (Yndira-E) [inline src/blog/2024/07/deploying-flowfuse-with-docker.md:5]: @sumitshinde-84 was it your plan to publish it next week?

Comment (sumitshinde-84) [reply src/blog/2024/07/deploying-flowfuse-with-docker.md:5]: feel free to update it !

Comment (sumitshinde-84) [reply src/blog/2024/07/deploying-flowfuse-with-docker.md:5]: Yes, I usually update dates once they receive at least one approval

**Pull request merged**